ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/replyboxview/ReplyBoxFragmentTest.java

```diff
@@ -0,0 +1,350 @@
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNotNull;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.CoreMatchers.is;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ErrorCollector;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.powermock.api.support.membermodification.MemberMatcher.methods;
+import static org.powermock.api.support.membermodification.MemberModifier.replace;
+import static org.powermock.api.support.membermodification.MemberModifier.suppress;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+
+@PrepareForTest({
+        KayakoCreditsHelper.class,
+        Intent.class,
+        Uri.class,
+        Fragment.class
+})
+@RunWith(PowerMockRunner.class)
+public class ReplyBoxFragmentTest {
+
+    private final ReplyBoxFragment replyBoxFragment = new ReplyBoxFragment();
+    private Uri uri;
+
+    @Mock
+    private Bundle bundle;
+
+    @Mock
+    private LayoutInflater layoutInflater;
+
+    @Mock
+    private ViewGroup viewGroup;
+
+    @Mock
+    private View view;
+
+    @Mock
+    private ReplyBoxContract.Presenter presenter;
+
+    @Mock
+    private EditText editText;
+
+    @Mock
+    private TextView textView;
+
+    @Mock
+    private KeyEvent keyEvent;
+
+    @Mock
+    private Intent intent;
+
+    @Mock
+    private AppCompatActivity appCompatActivity;
+
+    @Mock
+    private Editable editable;
+
+    @Mock
+    private ImageButton imageButton;
+
+    @Mock
+    private ReplyBoxContract.ReplyBoxListener listener;
+
+    @Captor
+    private ArgumentCaptor<String> stringArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<Integer> integerArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<Boolean> booleanArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<ReplyBoxContract.ReplyBoxListener> replyBoxListenerArgumentCaptor;
+
+    @Rule
+    public final ErrorCollector errorCollector = new ErrorCollector();
+
+    @Test
+    public void onCreate() {
+        //Act
+        replyBoxFragment.onCreate(bundle);
+        final ReplyBoxContract.Presenter mPresenter =
+                Whitebox.getInternalState(replyBoxFragment, "mPresenter");
```

Comment:
⚠️  This snippet of code uses so-called magic strings and sometimes is difficult to infer their goal in the application. It is a good practice to "extract" magic strings and provide meaningful names for them. In this case, consider extracting them into class-level symbolic constants. This approach increases maintainability (single point of modification) and debugging, avoid duplication and decrease verbosity.  

**[More details](https://confluence.devfactory.com/x/SI8-Ew)**
          
## The solution in this case

```Java  
    private static final String FIELD_M_PRESENTER = "mPresenter";
    // replace 4 usages of this literal with the new constant
```
 

<a id="b1591f60-53c9-11e8-b777-83ccc384eaae" href="none" codereview-uuid-reserved-tag />